File: Javascript/level 2/The longest Palindrome.js
function longest_palindrom(s){
  var result = 0;
  // 함수를 완성하세요
  for(var i=0;i<s.length;i++){
		for(var a=1; a<=s.length; a++){
  		if(s.substring(i,a)==s.substring(i,a).split("").reverse().join("")){
     	 if(result==0)
       	 result = s.substring(i,a).length;
     	 if(result < s.substring(i,a).length)
        	result = s.substring(i,a).length;
    	}
  	}
  }
  return result;
}


// 아래는 테스트로 출력해 보기 위한 코드입니다.
console.log( longest_palindrom("토마토맛토마토") )
console.log( longest_palindrom("토마토맛있어") )
console.log( longest_palindrom("나에게소주만병만주소.") )

File: Python/level 2/The longest Palindrome.py
def longest_palindrom(s):
    result = []
    for i in range(0,len(s)):
        for j in range(1,len(s)+1):
            if str(s[i:j])==str(s[i:j])[::-1]:
                result.append(len(s[i:j]))
            if str(s[j:i])==str(s[j:i])[::-1]:
                result.append(len(s[j:i]))
    return max(result)


# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(longest_palindrom("토마토맛토마토"))
print(longest_palindrom("토마토맛있어"))